test/run-tests.js

```diff
@@ -1,17 +1,34 @@
-const { readdirSync } = require('fs')
-const { join } = require('path')
 const npmRun = require('npm-run')
-const version = process.env.REACT || '16.4'
+const LATEST_VERSION = '16.5'
+const version = process.env.REACT || LATEST_VERSION
 
-try {
-  require('./install-test-deps.js')
-  if (version.toLowerCase() === 'all') {
-    readdirSync(join(__dirname, 'react')).forEach(version => {
-      npmRun.execSync(`cd test/react/${version} && npm test`, { stdio: 'inherit' })
-    })
-  } else {
-    npmRun.execSync(`cd test/react/${version} && npm test`, { stdio: 'inherit' })
+const jestConfig = {
+  testURL: 'http://localhost',
+  collectCoverage: true,
+  coverageDirectory: `${__dirname}/coverage`,
+  transform: {
+    '.js$': `${__dirname}/babel-transformer.jest.js`,
+  },
+}
+
+require('./install-test-deps.js')
+
+if (version.toLowerCase() === 'all') {
+  const allVersionsConfig = {
+    ...jestConfig,
+    rootDir: __dirname,
+    // every directory has the same coverage, so we collect it only from one
+    collectCoverageFrom: [`react/${LATEST_VERSION}/src/**.js`]
+  }
+  npmRun.execSync(`node ./node_modules/.bin/jest -c '${JSON.stringify(allVersionsConfig)}'`, { stdio: 'inherit' })
```

Comment:
it seems to me that `node ./node_modules/.bin/jest` can be used when testing a specific version too, combining this with `rootDir` option would allow us to remove usage of `npm-run` here, removing `cd` dance & removing 'artificial' test scripts from nested package.jsons 

WDYT?

Reply:
First, if we're using npm-run, then this should just be `jest -c ...`. That module automatically adds the .bin dir to the PATH.

We could actually make use of [`jest --projects`](https://jestjs.io/docs/en/configuration.html#projects-array-string-projectconfig) in this case. That might be easier, as each version of React would be testable in isolation and without as much tooling.

Reply:
> First, if we're using npm-run, then this should just be jest -c .... That module automatically adds the .bin dir to the PATH.

Cool, didnt know that - gonna chang it in a minute.

> We could actually make use of jest --projects in this case. That might be easier, as each version of React would be testable in isolation and without as much tooling.

I thought about it, but I think that would require keeping a copy of the jest config per each react version and is IMHO annoying in a long run (when it needs updating).

Also I wasnt sure how to collect coverage in `--projects` case as we wont to collect it only from a single project.

Reply:
Damn, it doesn't do per-project coverage? That seems like an oversight.

Reply:
I guess it might have, we just have exactly the same coverage here per project as noticed by @cellog. I guess overcollecting the coverage doesnt do us harm, but im not sure how CI takes the coverage for its report. Is the path configurable? Gonna look into that later as im on the mobile right now. If its configurable then i guess we might just point to the latest react directory and be done with it (leting jest collect in each directory).

There is still an issue of having to duplicate jest config all over the place though (if u want to use the projects feature)